examples/twittermap/web/public/javascripts/map/pointmap.js

```diff
@@ -253,20 +282,21 @@ angular.module('cloudberry.map')
               fillColor : '#b8e3ff',
               fillOpacity : 1.0
             }).addTo($scope.map);
-            //(3) Send request to twitter.com for the oembed json tweet content.
-            var url = "https://api.twitter.com/1/statuses/oembed.json?callback=JSON_CALLBACK&id=" + $scope.pointIDs[i];
-            $http.jsonp(url).success(function (data) {
-              var tweetContent = translateOembedTweet(data);
-              $scope.popUpTweet = L.popup({maxWidth:300, minWidth:300, maxHight:300});
-              $scope.popUpTweet.setContent(tweetContent);
-              $scope.currentMarker.bindPopup($scope.popUpTweet).openPopup();
-            }).
-            error(function() {
-              var tweetContent = "Sorry! It seems the tweet with that ID has been deleted by the author.@_@";
-              $scope.popUpTweet = L.popup({maxWidth:300, minWidth:300, maxHight:300});
-              $scope.popUpTweet.setContent(tweetContent);
-              $scope.currentMarker.bindPopup($scope.popUpTweet).openPopup();
-            });
+
+            //send the query to cloudberry.
+            var passID = "" + $scope.pointIDs[i];
+            cloudberry.pinMapOneTweetQuery(passID);
+
+            //monitors and receives the result with updating content of each pin tweet.
+              $scope.$watch(function () {
```

Comment:
Here, the $watch function should be called outside of the $scope.map.on function, because currently, everytime the mouse hang over a point, it will register the $watch variable and function once.